### PR TITLE
API-829 R4 allergy-intolerance controller

### PR DESCRIPTION
**Pull request by jblefkove-va**
https://vajira.max.gov/browse/API-558

Requires https://github.com/department-of-veterans-affairs/health-apis-fhir-resources/pull/58

### CODE REVIEW

File: data-query/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
   <packaging>jar</packaging>
   <properties>
     <fall-risk.version>1.0.29</fall-risk.version>
-    <fhir-resources.version>3.0.0</fhir-resources.version>
+    <fhir-resources.version>3.0.1-SNAPSHOT</fhir-resources.version>
```
Comment (bschofield-va) [inline data-query/pom.xml:15]: don't forget to rev this

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/metadata/MetadataSamples.java [hunk 2]
```diff
@@ -28,7 +28,6 @@ static MetadataProperties conformanceStatementProperties() {
                 + " It is provided to use as a template for system designers to"
                 + " build their own conformance statements from.")
         .softwareName("VA Lighthouse")
-        .fhirVersion("1.0.2")
```
Comment (bschofield-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/metadata/MetadataSamples.java:31]: why this?

Comment (jblefkove-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/metadata/MetadataSamples.java:31]: Wasn't this deprecated in favor of a hard-coded value in a previous PR?

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/R4AllergyIntoleranceControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+import javax.servlet.http.HttpServletResponse;
+import lombok.SneakyThrows;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@DataJpaTest
```
Comment (bschofield-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/allergyintolerance/R4AllergyIntoleranceControllerTest.java:33]: \* apprehensive noises *

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/metadata/MetadataSamples.java (hunk 2 above)
Comment (bschofield-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/metadata/MetadataSamples.java:31]: It's being set to that value in the dstu2  metadata controller explicitly, where it was previously injected via property. The `fhirVersion` field is still being set and should be continue to be set in the sample data.

Comment (bschofield-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/metadata/MetadataSamples.java:31]: https://github.com/department-of-veterans-affairs/health-apis-data-query/pull/501/files#diff-2b5427113570d8a4a582c73848d339baL186

**Pull request merged**